src/lib/calculate-colors.js

```diff
@@ -1,74 +1,192 @@
-const getColorYiq = (colorHex) => {
-  const r = parseInt(colorHex.substr(1, 2), 16)
-  const g = parseInt(colorHex.substr(3, 2), 16)
-  const b = parseInt(colorHex.substr(5, 2), 16)
-  const yiq = ((r * 299) + (g * 587) + (b * 114)) / 1000
-  return (yiq >= 128) ? 'var(--gray-dark)' : 'var(--white)'
-}
+// Dependencies
+const sass = require('sass')
 
-const getColorRgb = (colorHex) => {
-  // https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb
-  const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i
-  colorHex = colorHex.replace(shorthandRegex, function (m, r, g, b) {
-    return r + r + g + g + b + b
-  })
-  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(colorHex)
-  return result
-    ? `${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(result[3], 16)}`
-    : null
-}
+/**
+ *  Combile color range
+ * @param {string} primaryColor : Base primary color;
+ * @param {string} secondaryColor : Base secondary color;
+ * @returns {object} : Containing { css: sring, loadedUrls: Array }
+ */
+ const generateColorRangeSass = (primaryColor, secondaryColor) => {
 
-const darkenColor = (color, percent) => {
-  // https://css-tricks.com/snippets/javascript/lighten-darken-color/#comment-1754753
-  let usePound = false
-  const amt = percent < 0 ? percent * -5.25 : percent * -4
-  if (color[0] === '#') {
-    color = color.slice(1)
-    usePound = true
-  }
-  const num = parseInt(color, 16)
-  let r = (num >> 16) + amt
-  if (r > 255) r = 255
-  else if (r < 0) r = 0
-  let b = ((num >> 8) & 0x00FF) + amt
-  if (b > 255) b = 255
-  else if (b < 0) b = 0
-  let g = (num & 0x0000FF) + amt
-  if (g > 255) g = 255
-  else if (g < 0) g = 0
-  return (usePound ? '#' : '') + (g | (b << 8) | (r << 16)).toString(16)
-}
+  // Compile color variables
+  const compiledSass = sass.compileString(`
+    $primaryColor: ${primaryColor};
+    $secondaryColor: ${secondaryColor};
 
-const colorVariants = {
-  whiter: -75,
-  white: -50,
-  lightest: -33,
-  lighter: -21,
-  light: -10,
-  lighten: -7,
-  darken: 8,
-  dark: 10,
-  darker: 13,
-  darkest: 16,
-  black: 50
-}
+    @function yiq($color) {
+      $red : red($color);
+      $green: green($color);
+      $blue: blue($color);
+      $yiqValue: (($red*.299)+($green*.587)+($blue*.114));
+      
+      @if $yiqValue >= 128 {
+        @return var(--gray-dark)
+      } @else {
+        @return var(--white)
+      }
```

Comment:
```suggestion
      @if $yiqValue >= 150 {
        @return var(--gray-dark);
      } @else {
        @return var(--white);
      }
```

https://github.com/ecomplus/storefront/blob/master/%40ecomplus/storefront-twbs/scss/presets/_colors.scss#L188

Reply:
Alterado